internal/storage/v2/clickhouse/tracestore/config.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package tracestore
+
+import (
+	"time"
+
+	"github.com/asaskevich/govalidator"
+)
+
+type Config struct {
+	// Protocol is the protocol to use to connect to ClickHouse.
+	// Supported values are "native" and "http". Default is "native".
+	Protocol string `mapstructure:"protocol" valid:"in(native|http),optional"`
+	// Addresses contains a list of ClickHouse server addresses to connect to.
+	Addresses []string `mapstructure:"addresses" valid:"required"`
+	// Auth contains the authentication configuration to connect to ClickHouse.
+	Auth AuthConfig `mapstructure:"auth"`
+	// DialTimeout is the timeout for establishing a connection to ClickHouse.
+	DialTimeout time.Duration `mapstructure:"dial_timeout"`
+	// TODO: add more settings
+}
+
+type AuthConfig struct {
+	// Database is the ClickHouse database to connect to.
+	Database string `mapstructure:"database"`
+	// Username is the username to connect to ClickHouse.
+	Username string `mapstructure:"username"`
+	// Password is the password to connect to ClickHouse.
+	Password string `mapstructure:"password"`
```

Comment:
is basic auth the only way to connect? Usually we structure overall Auth as Basic+other types.

even better, we should really switch to using configauth and auth-n extensions

https://pkg.go.dev/go.opentelemetry.io/collector/config/configauth#section-readme

Reply:
I updated to use OTEL's config struct here. There's also JWT auth which I can add in a follow-up PR.